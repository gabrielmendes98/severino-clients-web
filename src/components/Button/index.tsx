import { ReactNode, FC } from 'react';

interface Props {
  children: ReactNode;
}

const Button: FC<Props> = ({ children, ...props }) => (
  <button {...props}>{children}</button>
);

export default Button;
