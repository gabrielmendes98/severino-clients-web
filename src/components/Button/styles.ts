import MuiButton from '@mui/material/Button';
import styled from '@emotion/styled';

export const BaseButton = styled(MuiButton)`
  text-transform: none;
`;
