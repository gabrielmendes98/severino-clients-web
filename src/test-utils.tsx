import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
