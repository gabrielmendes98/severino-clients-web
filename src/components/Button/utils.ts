import { ButtonProps } from '@mui/material/Button';

export const loadingSize = (size: ButtonProps['size']) => {
  switch (size) {
    case undefined:
      return 18;
    case 'small':
      return 18;
    case 'medium':
      return 22;
    case 'large':
      return 24;
  }
};
