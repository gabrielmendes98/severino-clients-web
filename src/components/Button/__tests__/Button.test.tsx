import { render, screen } from 'test-utils';
import Button from '../index';

it('should show loader when pass loading prop to button', () => {
  render(<Button loading>test</Button>);
  expect(screen.getByRole('progressbar')).toBeInTheDocument();
  expect(screen.queryByText('test')).not.toBeInTheDocument();
});

it('should not show loader when do not pass loading prop to button', () => {
  render(<Button>test</Button>);
  expect(screen.queryByRole('progressbar')).not.toBeInTheDocument();
  expect(screen.getByText('test')).toBeInTheDocument();
});

it('should match snapshot', () => {
  const { container } = render(<Button>default button</Button>);
  expect(container).toMatchSnapshot();
});
