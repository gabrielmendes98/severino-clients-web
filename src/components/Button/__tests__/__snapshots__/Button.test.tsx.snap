// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="Toastify"
  />
  <button
    class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root css-5hitqp-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    default button
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
